cli/src/util/systemVersions.ts

```diff
@@ -0,0 +1,37 @@
+import { execFile } from "./cp.js";
+import { ModusHomeDir } from "../custom/globals.js";
+
+export async function getGoVersion(): Promise<string | undefined> {
+  try {
+    const result = await execFile("go", ["version"], {
+      shell: true,
+      cwd: ModusHomeDir,
+      env: process.env,
+    });
+    const parts = result.stdout.split(" ");
+    const str = parts.length > 2 ? parts[2] : undefined;
+    if (str?.startsWith("go")) {
+      return str.slice(2);
+    }
+  } catch {}
+}
+
+export async function getTinyGoVersion(): Promise<string | undefined> {
+  try {
+    const result = await execFile("tinygo", ["version"], {
+      shell: true,
+      cwd: ModusHomeDir,
+      env: process.env,
+    });
+    const parts = result.stdout.split(" ");
+    return parts.length > 2 ? parts[2] : undefined;
+  } catch {}
+}
+
+export async function getNPMVersion(): Promise<string | undefined> {
+  try {
+    const result = await execFile("npm", ["--version"], { shell: true });
```

Comment:
Not sure if npm will care about the current directory, but it won't hurt.  However, we should probably pass through the env vars.  Maybe just put one `execOpts` at the top that all three functions can share?

Reply:
most likely it doesn't care, but let's pass in the env var at least